lib/Test/Mockify/Base.pm

```diff
@@ -0,0 +1,249 @@
+package Test::Mockify::Base;
```

Comment:
I actually started with Injector inheriting from Mockify. This worked really well except for one thing: Mockify has the getMockObject method, but that name does not make sense for the Injector. The only reason I introduced Base was so Mockify could keep getMockObject, while Injector has getVerifier instead.

Maybe this could be resolved by removing the dual-purpose of Mockify's mock object, which serves as a test double and the verification object. Another idea I had could be part of the solution: perhaps the Mockify/Injector itself could be the verification object. There would be a getVerifier method, and additionally for Mockify, the getMockObject method. The inheritance would have to be switched to make that work (Mockify inherits from Injector), if you desire to remove Base.

Injector              -->      Mockify
    getVerifier                     getMockObject

Base is meant to be abstract, such that it should not be instantiated itself.

Reply:
You really made me start to think about this. But I have this gut feeling that it is not a good idea to have the base package. I think the actual issue is that the Injector is an additional Functionality for Mockify. This addition should not change the main project and it should be implemented with this in mind. I want it in a way that I could replace this addition without changing the original project. ( Your addition is still amazing but I want to keep the things separated because its keep the project clean)

You could solve the "getMockObject" vs. "getVerifier" Issue like this:
base:
```perl
package basePackage;

sub new {
    my $class = shift;
    return bless({}, $class);
}

sub baseMethod {
    my $self = shift;
    return (caller(0))[3];
}

1;
```
child
```perl
package childPackage;
use base qw(basePackage);

sub new {
    my $class = shift;
    my $self = $class->SUPER::new();
    return bless({}, $class);
}

sub baseMethod {
    my $self = shift;
    die("'Please use 'sameButNotBaseMethod' instead of 'baseMethod'");
    return (caller(0))[3];
}
sub sameButNotBaseMethod {
    my $self = shift;
    return $self->SUPER::baseMethod();
}

1;
```
test ( use cases )
```perl
package baseTest;
use strict;
use FindBin;
use lib ($FindBin::Bin);
use parent 'TestBase';
use Test::More;

use basePackage;
use childPackage;
use Test::Exception;
#------------------------------------------------------------------------
sub testPlan{
    my $self = shift;
    my $BasePackage = basePackage->new();
    is($BasePackage->baseMethod(),'basePackage::baseMethod', 'prove call of base package method' );

    my $ChildPackage = childPackage->new();
    is($ChildPackage->sameButNotBaseMethod(),'basePackage::baseMethod', 'prove piped call to base package method' );

    throws_ok( sub { $ChildPackage->baseMethod() },
                   qr/Please use 'sameButNotBaseMethod' instead of 'baseMethod'/,
                   " prove the expected Error when sombody try to call the baseMethod of the child"
         );
    return;
}

__PACKAGE__->RunTest();
1;
```